src/lib/components/Pagination.svelte

```diff
@@ -0,0 +1,121 @@
+<script lang="ts">
+	import PaginationArrow from "./PaginationArrow.svelte";
+
+	interface PageLink {
+		isActive: boolean;
+		label: string;
+		pageIndex: number;
+	}
+
+	/**
+	 * The number of page links to show on either side of the current page link.
+	 */
+	const NUM_EXTRA_BUTTONS = 2;
+
+	const ellipsis: PageLink = {
+		isActive: false,
+		label: "...",
+		pageIndex: -1,
+	};
+
+	export let classNames = "";
+	export let numItemsPerPage: number;
+	export let numTotalItems: number;
+	export let onChange: (newPageIndex: number) => void = () => {};
+	export let pageIndex: number;
+
+	let pageLinks: PageLink[] = [];
+
+	const numTotalPages = Math.ceil(numTotalItems / numItemsPerPage);
+
+	$: {
```

Comment:
This seems a bit complex 👀

Reply:
copied from hub. I will see what should be done for simplification

Reply:
handled in https://github.com/huggingface/chat-ui/pull/772/commits/914dbd9a6023b2ff49d98748e8fbf2e4e5cbadd1